Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -1269,6 +1391,192 @@ nmos::channelmapping_activation_handler make_node_implementation_channelmapping_
+            { nmos::caps::format::color_sampling, nmos::make_caps_string_constraint({ sdp::samplings::YCbCr_4_2_2.name }) },
+            { nmos::caps::format::interlace_mode, nmos::make_caps_string_constraint({ nmos::interlace_modes::interlaced_bff.name, nmos::interlace_modes::interlaced_tff.name, nmos::interlace_modes::interlaced_psf.name }) },
+            { nmos::caps::format::colorspace, nmos::make_caps_string_constraint({ sdp::colorimetries::BT2020.name, sdp::colorimetries::BT709.name }) },
+            { nmos::caps::format::transfer_characteristic, nmos::make_caps_string_constraint({ sdp::transfer_characteristic_systems::SDR.name }) },
+            { nmos::caps::format::component_depth, nmos::make_caps_integer_constraint({}, 8, 12) },
+            { nmos::caps::transport::st2110_21_sender_type, nmos::make_caps_string_constraint({ sdp::type_parameters::type_N.name }) }
+        })
+    });
+
+    auto audio_sender_capabilities = value_of({
+        value_of({
+            { nmos::caps::format::media_type, nmos::make_caps_string_constraint({ nmos::media_types::audio_L(8).name, nmos::media_types::audio_L(16).name, nmos::media_types::audio_L(20).name, nmos::media_types::audio_L(24).name }) },
+            { nmos::caps::format::channel_count, nmos::make_caps_integer_constraint({ (int)impl::channels_repeat.size() }) },
+            { nmos::caps::format::sample_rate, nmos::make_caps_rational_constraint({ nmos::rational{48000, 1} }) },
+            { nmos::caps::format::sample_depth, nmos::make_caps_integer_constraint({ 8, 16, 20, 24 }) }
+        })
+    });
+
+    return [&gate, video_sender_capabilities, audio_sender_capabilities, video_sender_ids, audio_sender_ids](const nmos::id& sender_id, const web::json::value& active_constraints)
+    {
+        const bool video_found = video_sender_ids.end() != boost::range::find(video_sender_ids, sender_id);
+        const bool audio_found = audio_sender_ids.end() != boost::range::find(audio_sender_ids, sender_id);
+
+        const auto& sender_capabilities = video_found ? video_sender_capabilities : audio_found ? audio_sender_capabilities : throw std::logic_error("No Sender Capabilities found");
+
+        const auto& constraint_sets = nmos::fields::constraint_sets(active_constraints).as_array();
+        for (const auto& constraint_set : constraint_sets)
+        {
+            if (!nmos::caps::meta::enabled(constraint_set)) continue;
+            for (const auto& sender_caps_constraint_set : sender_capabilities.as_array())
+            {
+                if (nmos::experimental::is_constraint_subset(sender_caps_constraint_set, constraint_set, true) || // the Constraint Set makes Sender Constraint Set's constraints narrower
+                    nmos::experimental::is_constraint_subset(constraint_set, sender_caps_constraint_set)) // the Constraint Set is wider than the Sender Constraint Set so the latter is always compliant
```

Comment:
Hmm, I'm also not sure I understand what the `merge` parameter does...

Reply:
I feel confused each time I return to `constraints.{cpp,h}`, not sure this logic is 100% correct. Here we take Constraint Set A from Active Constraints and Constraint Set B from Sender Caps.

If B is narrower (allows less) than A (the 2nd half of the `if`), then we can accept such A because B cannot physically break wider constraints.

If B is wider (allows more) than A (the 1st half of the `if`), then we support everything that A requires, let's accept such A. `merge` here is needed to accept Constraint Sets that have less parameter constraints that Constraint Sets in Sender Caps. For instance, the user requests restrictions only on `frame_rate` and such a Constraint Set cannot be a subset of very restricted Sender Caps, so we need to merge this incoming parameter constraint into the existing Constraint Set and then check.

Reply:
I don't understand why the merge is necessary. Let's have a chat to discuss!